BaragonCore/src/main/java/com/hubspot/baragon/models/BaragonSource.java

```diff
@@ -0,0 +1,51 @@
+package com.hubspot.baragon.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class BaragonSource {
+  private final String name;
+  private final Optional<Integer> port;
+
+  @JsonCreator
```

Comment:
I might be misunderstanding part of this, but adding `BaragonSource.fromString()` means that Baragon will be able to deserialize a `BaragonSource` object from either a String *or* a JSON blob, so the explicit migration isn't required. If you added new fields to `BaragonSource` that don't have acceptable default values, then the migration would be necessary.